pkg/workers/sharings/share_data.go

```diff
@@ -529,7 +543,7 @@ func generateDirOrFilePatch(dirDoc *vfs.DirDoc, fileDoc *vfs.FileDoc) (*jsonapi.
 
 // getDocAtRecipient returns the revision of the document at the given
 // recipient.
-func getDocRevAtRecipient(doctype, docID string, recInfo *RecipientInfo) (string, error) {
+func getDocRevAtRecipient(newDoc *couchdb.JSONDoc, method string, doctype, docID string, recInfo *RecipientInfo) (string, error) {
```

Comment:
(probably nitpick) I find the name `getDocRevAtRecipient` to be no longer appropriate: this function doesn't just get the revision of the document at the recipient but also checks for changes in the context of an update… 
But that's for the sake of clarity.